File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/domain/usecase/ScanDevicesUseCase.kt
package com.maktubcompany.easy_blue_printer.plugin.domain.usecase

import com.maktubcompany.easy_blue_printer.plugin.domain.entities.BluetoothDeviceEntity
import com.maktubcompany.easy_blue_printer.plugin.domain.repository.BluetoothRepository

class ScanDevicesUseCase(private val repository: BluetoothRepository) {
    fun execute(): List<BluetoothDeviceEntity> {
        return repository.scanDevices()
    }
}

File: ios/Classes/plugin/domain/usecases/ScanDevicesUseCase.swift
import Foundation

public class ScanDevicesUseCase {
    private var repository: BluetoothRepository

    init(repository: BluetoothRepository) {
        self.repository = repository
    }

    public func execute(completion: @escaping ([BluetoothDeviceEntity]) -> Void) {
        repository.scanDevices { devices in
            completion(devices)
        }
    }
}
